data/src/main/java/app/tivi/data/repositories/shows/ShowsModule.kt

```diff
@@ -16,18 +16,92 @@
 
 package app.tivi.data.repositories.shows
 
+import app.tivi.data.daos.ShowImagesDao
+import app.tivi.data.daos.TiviShowDao
+import app.tivi.data.entities.ErrorResult
+import app.tivi.data.entities.Success
+import app.tivi.data.entities.TiviShow
 import app.tivi.inject.Tmdb
 import app.tivi.inject.Trakt
+import com.dropbox.android.external.store4.StoreBuilder
 import dagger.Binds
 import dagger.Module
+import dagger.Provides
+import javax.inject.Singleton
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+
+@Module(includes = [ShowsModuleBinds::class, ShowStoreModule::class])
+class ShowsModule
 
 @Module
-abstract class ShowsModule {
+internal abstract class ShowsModuleBinds {
     @Binds
     @Trakt
     abstract fun bindTraktShowDataSource(source: TraktShowDataSource): ShowDataSource
 
     @Binds
     @Tmdb
     abstract fun bindTmdbShowDataSource(source: TmdbShowDataSource): ShowDataSource
+
+    @Binds
+    @Tmdb
+    abstract fun bindTmdbShowImagesDataSource(source: TmdbShowImagesDataSource): ShowImagesDataSource
+}
+
+@Module
+class ShowStoreModule {
+    @Provides
+    @Singleton
+    fun provideShowStore(
+        showDao: TiviShowDao,
+        @Trakt traktShowDataSource: ShowDataSource,
+        @Tmdb tmdbShowDataSource: ShowDataSource
+    ): ShowStore {
+        return StoreBuilder.fromNonFlow { showId: Long ->
+            val localShow = showDao.getShowWithId(showId)
+                ?: throw IllegalArgumentException("No show with id $showId in database")
+
+            coroutineScope {
+                val traktResult = async {
```

Comment:
you "probably" want to do this merging when writing it to disk. i'm not well aware of this code but if you are just trying to update local value carefully (assuming if tmdbShowDataSource is just local), then it is better to do it before it is inserted to database.

Store considers the persister as source of truth so it will never return values from the fetcher as is.